app/scripts/modules/core/src/serverGroup/ServerGroupHeader.tsx

```diff
@@ -51,6 +51,65 @@ export class CloudProviderIcon extends React.Component<IServerGroupHeaderProps>
   }
 }
 
+export interface IImageListState {
+  collapsed: boolean;
+}
+
+export class ImageList extends React.Component<IServerGroupHeaderProps, IImageListState> {
+  private toggle() {
+    this.setState((previousState: IImageListState) => {
+      return { collapsed: !previousState.collapsed };
+    });
+  }
+
+  constructor(props: IServerGroupHeaderProps) {
+    super(props);
+    this.state = {
+      collapsed: true,
+    };
+    this.toggle = this.toggle.bind(this);
+  }
+
+  public render() {
+    const { images } = this.props;
+    const { collapsed } = this.state;
+
+    // if we only have 1 image, render as simple string.
```

Comment:
i think the code here does a decent job of self-describing.  this comment doesn't add much beyond it.